src/Mutation/FilteredCollection.php

```diff
@@ -47,6 +47,33 @@ public function set(array $data): void
         }
     }
 
+    public function getIterator()
+    {
+        $iterator = new \RecursiveIteratorIterator(new class($this->arguments) extends \RecursiveArrayIterator {
+            public function hasChildren(): bool
+            {
+                $current = $this->current();
+                if ($current instanceof Item) {
+                    return false;
+                }
+
+                if (is_array($current)) {
+                    return true;
+                }
+
+                foreach ($current->arguments as $argument) {
```

Comment:
If this is the only reason to make it public, we should encapsulate this logic into a method hasChild or something like this